internal/engine/wazevo/backend/isa/arm64/instr.go

```diff
@@ -659,6 +667,36 @@ func (i *instruction) asFpuMov128(rd, rn regalloc.VReg) {
 	i.rn, i.rd = operandNR(rn), operandNR(rd)
 }
 
+func (i *instruction) asMovToVec(rd, rn regalloc.VReg, arr vecArrangement, index vecIndex) {
+	i.kind = movToVec
+	i.rd = operandNR(rd)
+	i.rn = operandNR(rn)
+	i.u1 = uint64(arr)
+	i.u2 = uint64(index)
+}
+
+func (i *instruction) asMovFromVec(rd, rn regalloc.VReg, arr vecArrangement, index vecIndex) {
+	i.kind = movFromVec
+	i.rd = operandNR(rd)
+	i.rn = operandNR(rn)
+	i.u1 = uint64(arr)
+	i.u2 = uint64(index)
+}
+
+func (i *instruction) asVecRRR(op vecOp, rd, rn regalloc.VReg, arr vecArrangement) {
```

Comment:
this should take `rm` as well (see `aluRRR`).
```suggestion
func (i *instruction) asVecRRR(op vecOp, rd, rn, rm regalloc.VReg, arr vecArrangement) {
```